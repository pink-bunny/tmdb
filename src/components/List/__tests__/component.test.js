import React from 'react';
import { shallow } from 'enzyme';

import ListComponent from '../component';

describe('List component', () => {
  const requiredProps = {
    removeModalTxt: 'Mock modal text',
    listTitle: 'Mock list title',
    emptyListTxt: 'Mock empty list text',
    fetchList: jest.fn(),
    removeModalAction: jest.fn(),
    loading: true,
    error: '',
    totalItems: 21,
    currentPage: 1
  };
  const wrapper = shallow(<ListComponent {...requiredProps} />); /* eslint-disable-line react/jsx-props-no-spreading, max-len */

  it('matches snapshot with required props', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('matches snapshot with custom props', () => {
    wrapper.setProps({
      list: [{
        id: 1,
        title: 'Terminator: Dark Fate',
        overview: 'Decades after Sarah Connor prevented Judgment Day.',
        poster_path: '/vqzNJRH4YyquRiWxCCOH0aXggHI.jpg' /* eslint-disable-line camelcase */
      }]
    });
    expect(wrapper).toMatchSnapshot();
  });

  it('passes actions array to <MoviesList />', () => {
    const actionsList = wrapper.find('MoviesList').renderProp('actionsList')(1, 'Mock title');
    expect(actionsList).toMatchSnapshot();
  });
});
