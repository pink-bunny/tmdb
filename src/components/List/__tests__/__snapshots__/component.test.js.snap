// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List component matches snapshot with custom props 1`] = `
<Adapter>
  <Adapter>
    <Row
      gutter={0}
    >
      <Col
        offset={2}
        span={20}
      >
        <div
          className="top-margin"
        >
          <Title>
            Mock list title
          </Title>
        </div>
      </Col>
    </Row>
    <MoviesList
      actionsList={[Function]}
      currentPage={1}
      emptyListTxt="Mock empty list text"
      error=""
      list={
        Array [
          Object {
            "id": 1,
            "overview": "Decades after Sarah Connor prevented Judgment Day.",
            "poster_path": "/vqzNJRH4YyquRiWxCCOH0aXggHI.jpg",
            "title": "Terminator: Dark Fate",
          },
        ]
      }
      loading={true}
      onPaginationClick={[MockFunction]}
      totalItems={21}
    />
  </Adapter>
</Adapter>
`;

exports[`List component matches snapshot with required props 1`] = `
<Adapter>
  <Adapter>
    <Row
      gutter={0}
    >
      <Col
        offset={2}
        span={20}
      >
        <div
          className="top-margin"
        >
          <Title>
            Mock list title
          </Title>
        </div>
      </Col>
    </Row>
    <MoviesList
      actionsList={[Function]}
      currentPage={1}
      emptyListTxt="Mock empty list text"
      error=""
      list={null}
      loading={true}
      onPaginationClick={[MockFunction]}
      totalItems={21}
    />
  </Adapter>
</Adapter>
`;

exports[`List component passes actions array to <MoviesList /> 1`] = `
<WrappedComponentShell
  id={1}
  key="delete"
  removeModalAction={[MockFunction]}
  removeModalTxt="Mock modal text"
  title="Mock title"
/>
`;
